src/titiler/extensions/titiler/extensions/render.py

```diff
@@ -0,0 +1,123 @@
+"""render Extension."""
+
+from pprint import pprint
+from typing import Annotated, Dict, List, Optional
+from urllib.parse import urlencode
+
+from attrs import define
+from fastapi import Depends, HTTPException, Path, Request
+from fastapi.dependencies.utils import get_dependant, request_params_to_args
+from pydantic import BaseModel, RootModel
+
+from titiler.core.factory import FactoryExtension, MultiBaseTilerFactory
+
+
+class RenderItem(BaseModel):
+    """Render item for stac render extension."""
+
+    assets: List[str]
+    title: Optional[str] = None
+    rescale: Optional[list[Annotated[List[float], 2]]] = None
+    nodata: Optional[float] = None
+    colormap_name: Optional[str] = None
+    colormap: Optional[Dict] = None
+    color_formula: Optional[str] = None
+    resampling: Optional[str] = None
+    expression: Optional[str] = None
+    minmax_zoom: Optional[Annotated[List[int], 2]] = None
```

Comment:
I think you might want to allow more keys event if they are not supported directly by titiler